File: src/main/java/com/rtomyj/skc/model/HateoasLinks.java
package com.rtomyj.skc.model;

import java.util.List;

public interface HateoasLinks
{

    void setSelfLink();
    void setLinks();

    static <T extends HateoasLinks> void setLinks(List<T> instances)
    {
        instances.forEach(T::setLinks);
    }

}

File: src/main/kotlin/com/rtomyj/skc/model/HateoasLinks.kt
package com.rtomyj.skc.model

interface HateoasLinks {
    fun setSelfLink()
    fun setLinks()

    companion object {
        @JvmStatic
        fun setLinks(instances: List<HateoasLinks>) {
            instances
                .forEach { it.setLinks() }
        }
    }
}